src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -1091,92 +1060,31 @@ void CascadiaSettings::_WriteSettings(const std::string_view content, const hstr
-                                         nullptr) };
-
-    if (!hFile)
+    const auto pathToSettingsFile = _SettingsPath();
+    auto data = ReadUTF8FileIfExists(pathToSettingsFile);
+
+    if (!data)
     {
         // GH#5186 - We moved from profiles.json to settings.json; we want to
         // migrate any file we find. We're using MoveFile in case their settings.json
         // is a symbolic link.
-        std::filesystem::path pathToLegacySettingsFile{ std::wstring_view{ pathToSettingsFile } };
+        auto pathToLegacySettingsFile = pathToSettingsFile;
         pathToLegacySettingsFile.replace_filename(LegacySettingsFilename);
 
-        wil::unique_hfile hLegacyFile{ CreateFileW(pathToLegacySettingsFile.c_str(),
-                                                   GENERIC_READ,
-                                                   FILE_SHARE_READ | FILE_SHARE_WRITE,
-                                                   nullptr,
-                                                   OPEN_EXISTING,
-                                                   FILE_ATTRIBUTE_NORMAL,
-                                                   nullptr) };
+        data = ReadUTF8FileIfExists(pathToLegacySettingsFile);
 
-        if (hLegacyFile)
+        if (data)
         {
-            // Close the file handle, move it, and re-open the file in its new location.
-            hLegacyFile.reset();
-
             // Note: We're unsure if this is unsafe. Theoretically it's possible
             // that two instances of the app will try and move the settings file
             // simultaneously. We don't know what might happen in that scenario,
             // but we're also not sure how to safely lock the file to prevent
             // that from occurring.
-            THROW_LAST_ERROR_IF(!MoveFile(pathToLegacySettingsFile.c_str(),
-                                          pathToSettingsFile.c_str()));
-
-            hFile.reset(CreateFileW(pathToSettingsFile.c_str(),
-                                    GENERIC_READ,
-                                    FILE_SHARE_READ | FILE_SHARE_WRITE,
-                                    nullptr,
-                                    OPEN_EXISTING,
-                                    FILE_ATTRIBUTE_NORMAL,
-                                    nullptr));
-
-            // hFile shouldn't be INVALID. That's unexpected - We just moved the
-            // file, we should be able to open it. Throw the error so we can get
-            // some information here.
-            THROW_LAST_ERROR_IF(!hFile);
-        }
-        else
-        {
-            // If the roaming file didn't exist, and the local file doesn't exist,
-            //      that's fine. Just log the error and return nullopt - we'll
-            //      create the defaults.
-            LOG_LAST_ERROR();
-            return std::nullopt;
+            std::filesystem::rename(pathToLegacySettingsFile, pathToSettingsFile);

```

Comment:
`std::filesystem::rename` attempts to be atomic and we don't have to use Win32 for that.